### PR TITLE
Add Calculation functionality to calculator component

### PR DESCRIPTION
**Pull request by elassis**
# What I have Accomplished 🏁 :
- Add mathematic functionality to calculator ✔️
- Add class components ✔️
- Manipulated the state of the component using setState ✔️
- Use lifting-up state to share state and functions between child components ✔️ 
## Built With 🔨 👍 :
- React

### CODE REVIEW

File: math/src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+ */
+export default function calculate(obj, buttonName) {
+  if (buttonName === 'AC') {
+    return {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
```
Comment (Emmyn5600) [inline math/src/logic/calculate.js]: - [ ] Could you please remove commented code or lines in your code to look more clean and professional
calc5

File: math/src/logic/calculate.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    return { operation: buttonName };
+  }
+
+  // save the operation and shift 'next' into 'total'
+  return {
+    total: obj.next,
+    next: null,
+    operation: buttonName,
+  };
+}
```
Comment (Emmyn5600) [inline math/src/logic/calculate.js]: - [ ] Kindly, remove commented code or lines in your code to look more clean and professional

File: math/src/logic/operate.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  }
+  if (operation === 'x') {
+    return one.times(two).toString();
+  }
+  if (operation === '/') {
+    return one.div(two).toString();
+  }
+  if (operation === '%') {
+    return one.mod(two).toString();
+  }
```
Comment (Emmyn5600) [inline math/src/logic/operate.js]: - [ ] The % button should work as modulus, kindly implement that button as requested.

------------------------------------------------------Your Output--------------------------------
calc4

------------------------------------------------------ExpectedOutput--------------------------------
calc3
!

File: math/src/logic/operate.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    return one.plus(two).toString();
+  }
+  if (operation === '-') {
+    return one.minus(two).toString();
+  }
+  if (operation === 'x') {
+    return one.times(two).toString();
+  }
+  if (operation === '/') {
+    return one.div(two).toString();
```
Comment (Emmyn5600) [inline math/src/logic/operate.js]: - [ ] A number divisible by 0 should throw an error that is clear to everyone, kindly fix this issue.

-----------------------------------------------ExpectedOutput---------------------------------------
calc1

-----------------------------------------------YourOutput---------------------------------------

calc2

File: math/src/components/calculator.js [hunk 5]
```diff
@@ -1,44 +1,96 @@
...
+      }
+    }else{
+      this.text += value;
+    }
+    
+    if(obj.total || obj.total === null){
+      this.setState({
+        total:obj.total
+      })      
+    }
```
Comment (Emmyn5600) [inline math/src/components/calculator.js:44]: - [ ] Kindly, let us use descriptive message or way to display the error message to the users
calc6

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUESTED :recycle:

Hi, @elassis 

Great work on implementing the project requirements, however there are some issues to be solved before you get approved but you are almost there:muscle:. 
   
   - The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Changes required :writing_hand: 
 
      * #### Please check the comments under the review.
       
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by Dipeshtwis — APPROVED** [review]
# Status: Approved ✔️

## Hi @elassis, 

Good job on implementing the suggested changes. Well Done.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
---------

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
_**Dipesh Kumar**_

--------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**